src/cdk/a11y/key-manager/legacy-tree-key-manager.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Subject} from 'rxjs';
+import {
+  TREE_KEY_MANAGER,
+  TreeKeyManagerFactory,
+  TreeKeyManagerItem,
+  TreeKeyManagerStrategy,
+} from './tree-key-manager';
+
+/**
+ * @docs-private
+ *
+ * @deprecated LegacyTreeKeyManager deprecated. Use TreeKeyManager or inject a
+ * TreeKeyManagerStrategy instead. To be removed in a future version.
+ *
+ * @breaking-change 19.0.0
```

Comment:
That's correct, we added this class as an "opt-out" for backwards compatibility. Injecting the "opt-out" key manager will cause the tree not to manage it's own focus behavior and enables the `tabindex` Inputs.

In the long term, instead of using tabindex Input, we prefer that developers use a key manager.